app/src/main/java/de/danoeh/antennapod/activity/OnlineFeedViewActivity.java

```diff
@@ -505,13 +507,34 @@ private void handleUpdatedFeedStatus(Feed feed) {
             if (DownloadRequester.getInstance().isDownloadingFile(feed.getDownload_url())) {
                 viewBinding.subscribeButton.setEnabled(false);
                 viewBinding.subscribeButton.setText(R.string.subscribing_label);
+                if (UserPreferences.isEnableAutodownload()) {
+                    LinearLayout.LayoutParams layoutParams
+                            = (LinearLayout.LayoutParams) viewBinding.subscribeButton.getLayoutParams();
+                    layoutParams.setMargins(16, 16, 16, 0);
+                    viewBinding.subscribeButton.setLayoutParams(layoutParams);
+                    viewBinding.autoDownloadCheckBox.setVisibility(View.VISIBLE);
+                }
```

Comment:
Could you please move setting the visibility to one central location? I am having a hard time understanding why it needs to be modified here *and* in the `else` branch.